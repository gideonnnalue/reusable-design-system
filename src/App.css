* {
  margin: 0;
  padding: 0;
}

body {
  font-family: -apple-system, "Segoe UI", "Roboto", "Arial", sans-serif;
  color: #191d2f;
}
