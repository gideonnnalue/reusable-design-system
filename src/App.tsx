import { FC, useState } from "react";
import { ThemeProvider } from "@emotion/react";
import "./App.css";

import { SelectMenu } from "./components";
import theme from "./theme";

const { lightTheme, darkTheme } = theme;

const { Select, Option, OptionExtended, Divider } = SelectMenu;

const App: FC = () => {
  const [isDarkMode, setIsDarkMode] = useState(true);
  return (
    <ThemeProvider theme={isDarkMode ? darkTheme : lightTheme}>
      <div
        style={{
          backgroundColor: isDarkMode
            ? darkTheme.colors.black
            : lightTheme.backgroundColor,
          height: "100vh",
        }}
      >
        <Select defaultValue="" onChange={() => {}}>
          <OptionExtended
            value="hopper"
            desc="Grace Hopper was an American computer scientist and US navy rear admiral"
          >
            Hopper
          </OptionExtended>
          <OptionExtended
            value="holderton"
            desc="Frances Elizabeth Holberton was one of the programmers of the first computer"
          >
            Holberton
          </OptionExtended>
          <Divider />
          <OptionExtended
            value="teitelbaum"
            desc="Ruth Teitelbaum Hopper was one of the first computer programmers of the world"
          >
            Teitelbaum
          </OptionExtended>
        </Select>

        <div>
          <Select defaultValue="" onChange={() => {}}>
            <Option value="hopper">Hopper</Option>
            <Option value="holberton">Holberton</Option>
            <Divider />
            <Option value="antonelli">Antonelli</Option>
            <Option value="bartik">Bartik</Option>
            <Option value="teitelbaum">Teitelbaum</Option>
          </Select>
        </div>
      </div>
    </ThemeProvider>
  );
};

export default App;
