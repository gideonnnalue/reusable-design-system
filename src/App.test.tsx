import React from "react";
import { render, screen, cleanup } from "@testing-library/react";
import App from "./App";

afterEach(cleanup);

test("check tests", () => {
  console.log("working");
});

// test("renders learn react link", () => {
//   render(<App />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

// it("should take a snapshot", () => {
//   const { asFragment } = render(<App />);
//   expect(asFragment()).toMatchSnapshot();
// });
