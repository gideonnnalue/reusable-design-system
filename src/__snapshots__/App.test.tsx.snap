// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    <div>
      <div
        class="css-1hnyd0r"
        data-testid="select-wrapper"
      >
        <div
          class="css-fdvb0z"
          data-testid="select-button"
          role="button"
          tabindex="0"
        >
          <span
            data-testid="selected-text"
          >
            Select an Item
          </span>
          <img
            alt="arrow"
            src="chevron-down.svg"
          />
        </div>
        <div
          class="css-1vznqvx"
        >
          <ul
            class="css-5ubqti"
            role="listbox"
          >
            <li
              aria-disabled="false"
              aria-selected="false"
              class="css-k4di0t"
              data-testid="hopper"
              data-value="hopper"
              role="option"
              tabindex="-1"
            >
              <img
                alt="check"
                src="check-light.svg"
              />
              <span>
                Hopper
              </span>
            </li>
            <li
              aria-disabled="false"
              aria-selected="false"
              class="css-k4di0t"
              data-testid="holberton"
              data-value="holberton"
              role="option"
              tabindex="-1"
            >
              <img
                alt="check"
                src="check-light.svg"
              />
              <span>
                Holberton
              </span>
            </li>
            <div
              class="css-1trqa8a"
            />
            <li
              aria-disabled="false"
              aria-selected="false"
              class="css-k4di0t"
              data-testid="antonelli"
              data-value="antonelli"
              role="option"
              tabindex="-1"
            >
              <img
                alt="check"
                src="check-light.svg"
              />
              <span>
                Antonelli
              </span>
            </li>
            <li
              aria-disabled="false"
              aria-selected="false"
              class="css-k4di0t"
              data-testid="bartik"
              data-value="bartik"
              role="option"
              tabindex="-1"
            >
              <img
                alt="check"
                src="check-light.svg"
              />
              <span>
                Bartik
              </span>
            </li>
            <li
              aria-disabled="false"
              aria-selected="false"
              class="css-k4di0t"
              data-testid="teitelbaum"
              data-value="teitelbaum"
              role="option"
              tabindex="-1"
            >
              <img
                alt="check"
                src="check-light.svg"
              />
              <span>
                Teitelbaum
              </span>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
